services/slam/builtin/builtin.go

```diff
@@ -258,6 +259,29 @@ type AttrConfig struct {
 	Port             string            `json:"port"`
 }
 
+// Validate creates the list of implicit dependencies.
+func (config *AttrConfig) Validate(path string) ([]string, error) {
+	var deps []string
+
+	if config.Algorithm == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "algorithm")
+	}
+
+	if config.ConfigParams["mode"] == "" {
```

Comment:
Do we want to require config_params mode? In tech spec ([here](https://docs.google.com/document/d/1LdD_q20timS7nUF43Aqa1fJ__p8N1u8C-NVVT6C0V9I/edit?pli=1#heading=h.o94c2vdulprr)) we say no but if users do not implement it, it will fail future runtime checks when we compare algorithm to specified slam mode

Reply:
maybe we can make it required when we do the model stuff?

Reply:
misread stuff i think, fine with making it required here